DNN Platform/Library/Services/Search/Internals/LuceneControllerImpl.cs

```diff
@@ -274,41 +282,59 @@ public LuceneResults Search(LuceneSearchContext searchContext)
+                            BodySnippet = GetHighlightedText(highlighter, fieldQuery, searcher, match, Constants.BodyTag, searchContext.LuceneQuery.BodySnippetLength),
+                            DescriptionSnippet = GetHighlightedText(highlighter, fieldQuery, searcher, match, Constants.DescriptionTag, searchContext.LuceneQuery.TitleSnippetLength),
+                            TagSnippet = GetHighlightedText(highlighter, fieldQuery, searcher, match, Constants.Tag, searchContext.LuceneQuery.TitleSnippetLength),
+                            AuthorSnippet = GetHighlightedText(highlighter, fieldQuery, searcher, match, Constants.AuthorNameTag, searchContext.LuceneQuery.TitleSnippetLength),
+                            ContentSnippet = GetHighlightedText(highlighter, fieldQuery, searcher, match, Constants.ContentTag, searchContext.LuceneQuery.TitleSnippetLength)
+                        }).ToList();
                 return luceneResults;
+            }
+            catch (Exception ex)
+            {
+                if (retryTimes >= _searchRetryTimes)
+                {
+                    throw;
+                }
 
-            luceneResults.Results  = searchSecurityTrimmer.ScoreDocs.Select(match =>
-                new LuceneResult
-                    {
-                        Document = searcher.Doc(match.Doc),
-                        Score = match.Score,
-                        DisplayScore = GetDisplayScoreFromMatch(match.ToString()),
-                        TitleSnippet = GetHighlightedText(highlighter, fieldQuery, searcher, match, Constants.TitleTag, searchContext.LuceneQuery.TitleSnippetLength),
-                        BodySnippet = GetHighlightedText(highlighter, fieldQuery, searcher, match, Constants.BodyTag, searchContext.LuceneQuery.BodySnippetLength),
-                        DescriptionSnippet = GetHighlightedText(highlighter, fieldQuery, searcher, match, Constants.DescriptionTag, searchContext.LuceneQuery.TitleSnippetLength),
-                        TagSnippet = GetHighlightedText(highlighter, fieldQuery, searcher, match, Constants.Tag, searchContext.LuceneQuery.TitleSnippetLength),
-                        AuthorSnippet = GetHighlightedText(highlighter, fieldQuery, searcher, match, Constants.AuthorNameTag, searchContext.LuceneQuery.TitleSnippetLength),
-                        ContentSnippet = GetHighlightedText(highlighter, fieldQuery, searcher, match, Constants.ContentTag, searchContext.LuceneQuery.TitleSnippetLength)
-                    }).ToList();
-            return luceneResults;
+                DisposeReaders();
+                retryTimes++;
+                Logger.Error(ex);
+                Logger.Error($"Search Index Folder Is Not Available: {ex.Message}, Retry {retryTimes} time(s).");
+                Thread.Sleep(100);
+
+                return Search(searchContext, retryTimes);
```

Comment:
Please do not call `Search` recursively from `try..catch`. Instead please wrap error handling in a loop and break out of that loop when the results have been successfully returned.